packages/main/src/plugin/kubernetes/resource-informer.ts

```diff
@@ -0,0 +1,127 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { Informer, KubernetesObject, ListPromise, ObjectCache } from '@kubernetes/client-node';
+import { CHANGE, ERROR, ListWatch, Watch } from '@kubernetes/client-node';
+import type { Disposable } from '@podman-desktop/api';
+
+import type { Event } from '../events/emitter.js';
+import { Emitter } from '../events/emitter.js';
+import type { KubeConfigSingleContext } from './kubeconfig-single-context.js';
+
+interface BaseEvent {
+  kubeconfig: KubeConfigSingleContext;
+  resourceName: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
+export interface CacheUpdatedEvent extends BaseEvent {}
+
+export interface OfflineEvent extends BaseEvent {
+  offline: boolean;
+  reason?: string;
+}
+
+export class ResourceInformer<T extends KubernetesObject> implements Disposable {
```

Comment:
yes!

Reply:
In fact, the namespace is already part of the parameters, as part of the `path`.

For the reason we have discussed (being able to list a resource for several namespaces of a same cluster), I think that it would be better to inject these namespaces as part of the kubeconfig we pass to the `ContextManagerExperimental.update`, by duplicating the context and changing its namespace (either by modifying the user's kubeconfig file, or creating one in-memory for this), so we can do the entire process, including checking the permissions.

Reply:
maybe there was a previous discussion not shared before the 'add a namespace as parameter' but yes, I agree with the last comment, I think the scope is to have the namespace part of the configuration object that we're receiving, as it's for one context and a given namespace.
New namespace to monitor = new sub config to watch but using the same pattern/ logic.